sdk/storage/AzureStorageBlob/Source/Transport/URLSessionTransferManager+TransferDelegate.swift

```diff
@@ -50,17 +50,43 @@ extension URLSessionTransferManager: ResumableOperationQueueDelegate {
         default:
             if let blobTransfer = transfer as? BlobTransfer {
                 let progress = blobTransfer.progress
+
+                // avoid saving the context or sending multiple messages when the progress has not actually changed
                 if blobTransfer.currentProgress < progress {
                     blobTransfer.currentProgress = progress
                     delegate?.transfer(transfer, didUpdateWithState: state, andProgress: progress)
+                } else {
+                    return
                 }
             } else {
                 delegate?.transfer(transfer, didUpdateWithState: state)
             }
         }
+        saveContext()
+    }
+
+    func transfersDidUpdate(_ transfers: [Transfer]) {
+        guard let restorationId = (transfers.first as? TransferImpl)?.clientRestorationId else { return }
```

Comment:
There's an assumption here that all transfers in `transfers` come from the same client. Is that always true? If not we could refactor this to group transfers by their restoration ID, and then notify each client with the appropriate subset of `transfers`.

Reply:
Good point.